src/main/java/ca/corefacility/bioinformatics/irida/web/controller/api/samples/RESTSampleSequenceFilesController.java

```diff
@@ -469,7 +474,10 @@ public ModelMap addNewSequenceFileToSample(@PathVariable Long sampleId, @Request
 
 		} catch (IllegalArgumentException e) {
 			logger.debug("Error 400 - Bad Request: " + e.getMessage());
-			response.setStatus(HttpStatus.BAD_REQUEST.value());
+			throw e;
+		} finally {
```

Comment:
Thanks for adding the finally here.  This should (hopefully) help with some messes left behind from uploads.